### PR TITLE
refactor: HDInsights new ID generators

### PR DESCRIPTION
**Pull request by favoretti**
My subs weren't ready to run acceptance tests on core limits unfortunately.

### CODE REVIEW

File: azurerm/internal/services/hdinsight/parse/hd_insight_cluster.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	}
+}
+
+func (id HDInsightClusterId) String() string {
+	segments := []string{
+		fmt.Sprintf("Cluster Name %q", id.ClusterName),
+		fmt.Sprintf("Resource Group %q", id.ResourceGroup),
+	}
+	segmentsStr := strings.Join(segments, " / ")
+	return fmt.Sprintf("%s: (%s)", "H D Insight Cluster", segmentsStr)
```
Comment (tombuildsstuff) [inline azurerm/internal/services/hdinsight/parse/hd_insight_cluster.go:32]: that's a generator bug we should probably fix 🤔

File: azurerm/internal/services/hdinsight/resourceids.go [hunk 2]
```diff
@@ -0,0 +1,3 @@
+package hdinsight
+
+//go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=HDInsightCluster -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.HDInsight/clusters/cluster1
```
Comment (tombuildsstuff) [inline azurerm/internal/services/hdinsight/resourceids.go:3]: in general we could probably ditch the repeated package name here, e.g. `HDInsight Cluster` could become `Cluster`:

```suggestion
//go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=Cluster -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.HDInsight/clusters/cluster1
```

meaning that the field `ClusterName` in the generated ID Struct would become `Name` - not a big deal just a heads up :)

**Review by tombuildsstuff — APPROVED** [review]
hey @favoretti 

I've taken a look through and left a couple of minor comments (one a bug, one a heads-up) but this otherwise looks good to me 👍

Thanks!

File: azurerm/internal/services/hdinsight/resourceids.go (hunk 2 above)
Comment (favoretti) [reply azurerm/internal/services/hdinsight/resourceids.go:3]: I'll regenerate with this one and change it.

File: azurerm/internal/services/hdinsight/parse/hd_insight_cluster.go (hunk 1 above)
Comment (favoretti) [reply azurerm/internal/services/hdinsight/parse/hd_insight_cluster.go]: "Fixed" by renaming 👀

**Comment by tombuildsstuff** [comment]
Tests look good 👍

**Pull request merged**